zookeeper-server/src/test/java/org/apache/zookeeper/common/X509UtilTest.java

```diff
@@ -538,6 +540,7 @@ public void testClientRenegotiationFails() throws Throwable {
         SSLSocket clientSocket = null;
         SSLSocket serverSocket = null;
         final AtomicInteger handshakesCompleted = new AtomicInteger(0);
```

Comment:
Looks like we can remove handshakesCompleted after adding the count down.

Reply:
I thought about it.

The comment below states that we want to check that the handshake completes only once and not twice.

Now, the check is performed in the finally block, and the game should be over at that point.

This change slows down the test by forcing a synchronization point.

As we are waiting only for the first call we could still have a race and miss the second call but at least we are not changing the core logic of this test

Reply:
So I would like to keep the AtomicInteger